extmod/modwebsocket.c

```diff
@@ -58,16 +63,39 @@ typedef struct _mp_obj_websocket_t {
 STATIC mp_uint_t websocket_write(mp_obj_t self_in, const void *buf, mp_uint_t size, int *errcode);
 
 STATIC mp_obj_t websocket_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 1, 2, false);
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_sock, MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_use_blocking_writes, MP_ARG_BOOL, {.u_bool = false} },
+        { MP_QSTR_is_client, MP_ARG_BOOL, {.u_bool = false} },
+        { MP_QSTR_debug_mask, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
```

Comment:
I'm always for having debugging/diagnostic facilities in the code, but they should be properly #ifdef'ed and off by default.